subsys/net/lib/coap/coap_transport_socket.c

```diff
@@ -407,7 +417,7 @@ u32_t coap_security_setup(coap_local_t *local, struct sockaddr const *remote)
 						  remote,
 						  address_length_get(remote));
 
-				if (err) {
+				if (err && errno != EINPROGRESS) {
```

Comment:
And what about `sendto`/`recvfrom`? They also need special treatment in case of non-blocking socket.

Reply:
For ```recvfrom()``` we already set ```MSG_DONTWAIT``` so that should be covered. For ```sendto()``` I'll have a closer look.

Reply:
Actually we only need a non-blocking ```connect()``` during ```coap_security_setup()```. I have updated the code.